consensus/polybft/stake_manager_fuzz_test.go

```diff
@@ -0,0 +1,288 @@
+			Validators: newValidatorStakeMap(validators.GetPublicIdentities(initialSetAliases...)),
+		}))
+
+		receipt := &types.Receipt{
+			Logs: []*types.Log{
+				createTestLogForTransferEvent(
+					t,
+					stakeManager.validatorSetContract,
+					validators.GetValidator(initialSetAliases[data.ValidatorID]).Address(),
+					types.ZeroAddress,
+					data.StakeValue,
+				),
+			},
+		}
+
+		req := &PostBlockRequest{
+			FullBlock: &types.FullBlock{Block: &types.Block{Header: &types.Header{Number: data.BlockID}},
+				Receipts: []*types.Receipt{receipt},
+			},
+			Epoch: data.EpochID,
+		}
+		err := stakeManager.PostBlock(req)
+		require.NoError(t, err)
+	})
+}
+
+func FuzzTestStakeManagerUpdateValidatorSet(f *testing.F) {
+	var (
+		aliases = []string{"A", "B", "C", "D", "E"}
+		stakes  = []uint64{10, 10, 10, 10, 10}
+	)
+
+	validators := validator.NewTestValidatorsWithAliases(f, aliases, stakes)
+	state := newTestState(f)
+
+	stakeManager := newStakeManager(
+		hclog.NewNullLogger(),
+		state,
+		nil,
+		wallet.NewEcdsaSigner(validators.GetValidator("A").Key()),
+		types.StringToAddress("0x0001"), types.StringToAddress("0x0002"),
+		10,
+	)
+
+	seeds := []updateValidatorSetF{
+		{
+			EpochID:     0,
+			X:           1,
+			VotingPower: 30,
+		},
+		{
+			EpochID:     1,
+			X:           4,
+			VotingPower: 1,
+		},
+		{
+			EpochID:     2,
+			X:           3,
+			VotingPower: -2,
+		},
+	}
+
+	for _, seed := range seeds {
+		data, err := json.Marshal(seed)
+		if err != nil {
+			return
+		}
+
+		f.Add(data)
+	}
+
+	f.Fuzz(func(t *testing.T, input []byte) {
+		var data updateValidatorSetF
+		if err := json.Unmarshal(input, &data); err != nil {
+			t.Skip(err)
+		}
+
+		if err := ValidateStruct(data); err != nil {
+			t.Skip(err)
+		}
+
+		if data.X > uint64(len(aliases)-1) {
+			t.Skip()
+		}
+
+		err := state.StakeStore.insertFullValidatorSet(validatorSetState{
+			Validators: newValidatorStakeMap(validators.GetPublicIdentities())})
+		require.NoError(t, err)
+
+		_, err = stakeManager.UpdateValidatorSet(data.EpochID, validators.GetPublicIdentities(aliases[data.X:]...))
+		require.NoError(t, err)
+
+		fullValidatorSet := validators.GetPublicIdentities().Copy()
+		validatorToUpdate := fullValidatorSet[data.X]
+		validatorToUpdate.VotingPower = big.NewInt(data.VotingPower)
+
+		_, err = stakeManager.UpdateValidatorSet(data.EpochID, validators.GetPublicIdentities())
+		require.NoError(t, err)
+	})
+}
+
+func ValidateStruct(s interface{}) (err error) {
+	structType := reflect.TypeOf(s)
+	if structType.Kind() != reflect.Struct {
+		return errors.New("input param should be a struct")
+	}
+
+	structVal := reflect.ValueOf(s)
+	fieldNum := structVal.NumField()
+
+	for i := 0; i < fieldNum; i++ {
+		field := structVal.Field(i)
+		fieldName := structType.Field(i).Name
+
+		isSet := field.IsValid() && !field.IsZero()
+
+		if !isSet {
+			err = fmt.Errorf("%w%s in not set; ", err, fieldName)
```

Comment:
I would leave all the errors.

Reply:
It's not necessary. I left it that way to get the info on what fields we have problems with because the initial idea was to validate the entire struct, but I found issues when validating a struct that has an array field.
Should I then leave all errors, or return when the first one happens?

Reply:
My opinion is that first error is enough, Nemanja doesn't agree, so your call, do it however you like better (there are not too many fields, so looping through each one of it would not harm) :slightly_smiling_face: 

Just please fix this typo which I stated in the suggestion (`in` instead of `is` in the error msg) :pray: